plugins/optimization-detective/detect.js

```diff
@@ -335,6 +336,64 @@ function extendElementData( xpath, properties ) {
 	}
 	const elementData = elementsByXPath.get( xpath );
 	Object.assign( elementData, properties );
+	debouncedCompressUrlMetric();
+}
+
+/**
+ * Compresses a (JSON) string using CompressionStream API.
+ *
+ * @param {string} jsonString - JSON string to compress.
+ * @return {Promise<Blob>} Compressed data.
+ */
+async function compress( jsonString ) {
+	const encodedData = new TextEncoder().encode( jsonString );
+	const compressedDataStream = new Blob( [ encodedData ] )
+		.stream()
+		.pipeThrough( new CompressionStream( 'gzip' ) );
+	const compressedDataBuffer = await new Response(
+		compressedDataStream
+	).arrayBuffer();
+	return new Blob( [ compressedDataBuffer ], { type: 'application/gzip' } );
+}
+
+/**
+ * Stores the compressed URL metric data.
+ *
+ * @type {Blob|null}
+ */
+let compressedPayload = null;
+
+/**
+ * Timeout ID for debouncing URL metric compression.
+ *
+ * @type {ReturnType<typeof setTimeout> | null}
```

Comment:
Is this any different than the following?

```suggestion
 * @type {number|null}
```

Reply:
Yes its different TypeScript throws as error if we only put `number`
> Type 'Timeout' is not assignable to type 'number'
```js
recompressionTimeout = setTimeout( async () => {
```

Reply:
Ah, good to know.